docs/language/builtins.rst

```diff
@@ -311,6 +311,35 @@ Its underlying type is ``bytes32``, but it will be reported correctly as the ``H
 .. include:: ../examples/substrate/hash_type.sol
   :code: solidity
 
+chain_extension(uint32 ID, bytes input) returns (uint32, bytes)
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Only available on Substrate. Call the chain extension with the given ``ID`` and ``input`` data.
+Returns the return value from the chain extension and the output data.
+
+This function is a low level interface; the input and output data buffers are just ``bytes``.
+The caller is responsible for encoding the input and decoding the output correctly.
+We expect parachain authors to write their own higher level libraries on top.
+
+.. warning::
+    This function calls the runtime API `call_chain_extension <https://docs.rs/pallet-contracts/latest/pallet_contracts/api_doc/trait.Version0.html#tymethod.call_chain_extension>`.
+	It assumes that the implementation of the chain extension does
+	- read the input from the ``input_ptr`` parameter, used as a buffer pointer
+	- write potential output into the buffer found at the ``output_ptr`` pointer
+	- respect the output buffer length in ``output_len_ptr`` to prevent OOB writes. The output buffer is 16KB in size.
+	- write the amount of bytes written to ``output_ptr`` into the buffer at ``output_len_ptr``
+	
+	Unlike with other runtime API calls, the contracts pallet can not guarantee this behaviour.
+	Instead, it's specific to each chain extension. Hence, before using this builtin,
+	you must make sure that the chain extension being called is compatible.
```

Comment:
How about replacing
`Hence, before using this builtin, you must make sure that the chain extension being called is compatible. `
with
`Hence, when using this builtin, ensure that the chain extension is used in the correct way.`

Reply:
This suggests that the chain extension function / ID of interest can be used in various ways, and the caller can do something about that, doesn't it? But this is not the case. Either the implementation of the chain extension adheres to this, or it doesn't. There is nothing you can do about it (unless your are also the author of the parachain runtime, but this is a bit of a stretch).

My main goal of this warning is to make contract authors to reconsider whether they know what they are doing. Calling into a chain extension that you're not 100% certain about what it does is a really, really bad idea anyways.